File: repositorioacademico.client/src/app/models/usuario.ts
import { RolResumen } from './rol';

export interface Usuario {
  id: number;
  nombres: string;
  apellidos: string;
  correo: string;
  carnet: string;
  estado: string;
  fechaCreacion: string;
  roles: RolResumen[];
  permisos: string[];
}

File: RepositorioAcademico.Server/Domain/Usuario.cs
namespace RepositorioAcademico.Server.Domain
{
    public class Usuario
    {
        public int Id { get; set; }

        public string Nombres { get; set; } = string.Empty;

        public string Apellidos { get; set; } = string.Empty;

        public string Correo { get; set; } = string.Empty;

        public string Carnet { get; set; } = string.Empty;

        public string PasswordHash { get; set; } = string.Empty;

        public string Estado { get; set; } = "Activo";

        public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;

        public ICollection<Documento> Documentos { get; set; } = [];

        public ICollection<UsuarioRol> UsuarioRoles { get; set; } = [];
    }
}
